src/node/module.ts

```diff
@@ -21,16 +24,27 @@ export function createRequire(
     );
   }
 
-  return moduleUtil.createRequire(normalizedPath);
+  // TODO(soon): We should move this to C++ land.
+  // We do not add extensions field because it is deprecated.
+  // Ref: https://nodejs.org/docs/latest/api/modules.html#requireid
+  return Object.assign(moduleUtil.createRequire(normalizedPath), {
+    resolve(): void {
+      throw new ERR_METHOD_NOT_IMPLEMENTED('require.resolve()');
+    },
+    get cache(): void {
+      throw new ERR_METHOD_NOT_IMPLEMENTED('require.cache');
+    },
+    get main(): void {
+      throw new ERR_METHOD_NOT_IMPLEMENTED('require.main');
+    },
```

Comment:
For better readability, it would be helpful to move these function bodies out and simply refer to them here.